### PR TITLE
ConsolePlugin resource

### PR DESCRIPTION
**Pull request by jasonmadigan**
WIP

Ref: https://github.com/Kuadrant/console-plugin/issues/46

On OCP/OKD, apply a `ConsolePlugin` resource and associated deployment, to deploy https://github.com/Kuadrant/console-plugin alongside the rest of our components.

### CODE REVIEW

File: bundle/manifests/kuadrant-console-plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,133 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kuadrant-console
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kuadrant-console
+  namespace: kuadrant-console
```
Comment (eguzki) [inline bundle/manifests/kuadrant-console-plugin.yaml:10]: this namespace is quite arbitrary. I would try to make it configurable at operator build time (or runtime.. let's discuss about this). One candidate for the default value would be the namespace where the operator is installed. But it is ok as well to be in `kuadrant-console`.

File: controllers/kuadrant_controller.go [hunk 2]
```diff
@@ -167,6 +179,17 @@ func (r *KuadrantReconciler) Reconcile(eventCtx context.Context, req ctrl.Reques
 		return statusResult, nil
 	}
 
+	if r.isConsolePluginAvailable() {
+		logger.Info("ConsolePlugin API is available, proceeding to apply manifest")
+		if err := r.applyManifest(ctx, "bundle/manifests/kuadrant-console-plugin.yaml", r.Client()); err != nil {
```
Comment (eguzki) [inline controllers/kuadrant_controller.go:184]: I am afraid, this is not gonna work 

I see four approaches:
* The kuadrant operator binary has embedded the manifests. (we use this approach for the grafana dashboards on the 3scale operator)
* The kuadrant operator reads from a configmap
* The kuadrant operator has the manifests available in the docker image, so the container can read the file from local filesystem as you try there. 
* The kuadrant operator creates deployment, service, nginx configmap... So all those resources are encoded in golang code. Pretty much like the limitador operator does with limitador. 

Each approach has pro's & con's. Maybe a combination of some of them. IDK.  I would go first with the fourth approach. But let's discuss.

File: controllers/kuadrant_controller.go [hunk 3]
```diff
@@ -479,3 +502,58 @@ func (r *KuadrantReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&authorinov1beta1.Authorino{}).
 		Complete(r)
 }
+
+func (r *KuadrantReconciler) isConsolePluginAvailable() bool {
```
Comment (eguzki) [inline controllers/kuadrant_controller.go:506]: It is already implemented in `kuadrantgatewayapi.IsCRDInstalled`
I patched it and the outcome is:
```diff
diff --git a/controllers/kuadrant_controller.go b/controllers/kuadrant_controller.go
index b7c89dd..7d7589f 100644
--- a/controllers/kuadrant_controller.go
+++ b/controllers/kuadrant_controller.go
@@ -177,7 +177,13 @@ func (r *KuadrantReconciler) Reconcile(eventCtx context.Context, req ctrl.Reques
                return statusResult, nil
        }
 
-       if r.isConsolePluginAvailable() {
+       consolePluginGVK := consolev1.GroupVersion.WithKind("ConsolePlugin")
+       isConsolePluginAvailable, err := kuadrantgatewayapi.IsCRDInstalled(r.RestMapper, consolePluginGVK.Group, consolePluginGVK.Kind, consolePluginGVK.Version)
+       if err != nil {
+               return ctrl.Result{}, err
+       }
+
+       if isConsolePluginAvailable {
                logger.Info("ConsolePlugin API is available, proceeding to apply manifest")
                if err := r.applyManifest(ctx, "bundle/manifests/kuadrant-console-plugin.yaml", r.Client()); err != nil {
                        logger.Error(err, "failed to apply manifest")
@@ -501,17 +507,6 @@ func (r *KuadrantReconciler) SetupWithManager(mgr ctrl.Manager) error {
                Complete(r)
 }
 
-func (r *KuadrantReconciler) isConsolePluginAvailable() bool {
-       gvk := consolev1.GroupVersion.WithKind("ConsolePlugin")
-       mapping, err := r.RestMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-       if err != nil || mapping == nil {
-               log.Log.Info("ConsolePlugin API is not available")
-               return false
-       }
-       log.Log.Info("ConsolePlugin API is available")
-       return true
-}
-
 func (r *KuadrantReconciler) applyManifest(ctx context.Context, filePath string, k8sClient client.Client) error {
        logger := logr.FromContextOrDiscard(ctx)
```

**Review by eguzki — COMMENTED** [review]
Sweet, very cool stuff.

Left some comments we can use for further discussion

**Comment by jasonmadigan** [comment]
superseded by #884

**Pull request closed without merging**